### PR TITLE
Added Settings Page with Scalability Features

### LINKED ISSUE #100: Settings Page
## Summary
Make the settings page not use the generalInfo component.
You can choose how you want this settings page to look and function.
Free reign in design and most functionality

## Technical Details
Client, maybe Server
If you are making changes to a user then you will need to look into changing the update user route in the server (PUT endpoint)
You will need to add to the user.slice file in the store folder to handle this update

## Acceptance Criteria
- This is a free reign task since we have no figma for it
- You can choose what a users settings are and we can revisit it later
- Call a PUT endpoint to update user info if needed

## Figma
No Figma for this. Figure something out, you have free reign. We currently have the following:
image

### PR DESCRIPTION
**Pull request by morenj**
## Description
This pull request introduces a new settings page that allows us to add customization to the user's account.
Currently, this PR only adds the layout and a path to implement two components to update a user's profile and their password. 

## Related Issue
Closes #100

## Type of Changes
- [x] New feature

## Things to Update
In future updates, we might need to address the following items to enhance the settings page:
- [ ] Update Profile Information component
- [ ] Update Password component
- [ ] Code Documentation
- [ ] Improved Mobile Responsiveness
- [ ] Code Testing: Increase test coverage for the new settings page functionalities.

## Screenshots
*Settings Page Interface*
image
image

## Additional Notes
Reviewers should note that the settings page is designed to be scalable, allowing for easy inclusion of additional settings as needed. Testing on different devices for responsiveness and functionality is recommended. Further feedback on UI/UX improvements would be greatly appreciated to enhance user interaction.

### CODE REVIEW

**Review by tbmcallister — APPROVED** [review]
Page looks great, and I think this’ll greatly help our MVP be completed by the due date. Great work Juan!

**Review by B-Timok — APPROVED** [review]
Everything checks out for me. Nice to get closer to our MVP with this. Good Job!

File: client/src/pages/SettingsPage/SettingsPage.module.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        justify-content: flex-start;
+        align-items: stretch;
+        border-radius: 10px;
+        box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+        max-height: calc(100vh - 250px);
+        overflow-y: auto;
+
+        @media (max-width: 768px) {
+            width: 100%;
+        }
```
Comment (RedPenguin88) [inline client/src/pages/SettingsPage/SettingsPage.module.scss:109]: Once again, just converting the `px` values to `rem`. Feel free to adjust the values to your liking.
```suggestion
        height: 100%;
        padding-left: 1.25rem;
        padding-right: 1.25rem;
        padding-top: 0.75rem;
        justify-content: flex-start;
        align-items: stretch;
        border-radius: 0.75rem;
        box-shadow: 0 0 0.75rem rgba(0, 0, 0, 0.1);
        max-height: calc(100vh - 15.75rem);
        overflow-y: auto;

        @media (width <= 48.00rem) {
            width: 100%;
        }
```

**Review by RedPenguin88 — APPROVED** [review]
Great Work! This is essential for our MVP and does a great deal to bolster our project. I have a couple suggestions but I don't want that to hinder the merging process. Just a couple of things to think about for the future.

I know it seems like a lot, but I've provided suggestions that you should be able to commit with no problem whatsoever.

**Pull request merged**